var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -130,6 +130,10 @@ class PyNumpy(PythonPackage):
     patch('check_executables4.patch', when='@1.14.0:1.15.4')
     patch('check_executables5.patch', when='@:1.13.3')
 
+    # Backport bug fix for f2py's define for threading when building with Mingw
+    # to 1.20.3:1.22.1 - https://github.com/numpy/numpy/pull/20881/commits
+    patch('0001-BUG-fix-f2py-s-define-for-threading-when-building-wi.patch', when='@1.20.3:1.22.1')
```

Comment:
```suggestion
    patch('https://github.com/numpy/numpy/pull/20881.patch?full_index=1', 
          sha256='802970a9034d40a8a8f49a03f489d5361d5eabf69249621e6757651448910f1a', 
          when='@1.20.3:1.22.1')
```
Then you don't have to store the actual patch file in Spack. We've been trying to push for this more often lately because it isn't always clear what the license for patch files is and whether they can be stored in Spack.

Reply:
Thanks, that's a great idea.

Reply:
Interesting. For my education: doesn't this give problems with builds in isolated environments, or when internet is down? Is there a way to download and cache patches in bulk before going offline?

Reply:
I guess that is a question for @adamjstewart

Reply:
It doesn't present any more problems than the fact that the package itself also downloads from a URL. For users on air-gapped networks, you can use `spack mirror` to automatically download all tarballs and patches on a system with internet access, then copy them to a system without: https://spack.readthedocs.io/en/latest/mirrors.html

Reply:
That's what I thought, too, I just wasn't sure if patches that are not files in the spack repo are treated differently. Thanks for clarifying.

Reply:
Makes sense, thanks!